sklearn/utils/_array_api.py

```diff
@@ -1101,3 +1101,21 @@ def _tolist(array, xp=None):
         return array.tolist()
     array_np = _convert_to_numpy(array, xp=xp)
     return [element.item() for element in array_np]
+
+
+def _nan_to_num(array, xp=None):
+    """Substitutes NaN values with 0 and inf values with the maximum or minimum
+    numbers available for the dtype respectively; like np.nan_to_num."""
+    if xp is None:
+        xp, _ = get_namespace(array, xp=xp)
```

Comment:
```suggestion
    xp, _ = get_namespace(array, xp=xp)
```

Reply:
Yes, thank you @OmarManzoor and @ogrisel. I have deleted the condition.

But I'm wondering: Why we allow `xp` as an input into the helper functions in this file and then still get `xp` anew using `get_namespace()`. It seems redundant and maybe it could also hide subtle bugs in the case where our array has changed its namespace during the course of the execution of the actual function (the metric in this case) and we think we pass a different one into this function using `xp=xp` than we actually do, but in fact the array's namespace gets re-evaluated and corrected and it's true namespace is used. I hope I managed to express what I mean here.

Maybe we should not allow the helper functions to have an xp passed or we don't re-evaluate by getting the namespace anew, just so that it's always definite and easily visible which namespace is used?

Reply:
When we pass xp to this function it just returns it back. I agree it might be better to ignore calling namespace again when we already have xp but I think this is the convention being used in this utils file and it makes the code usable from a place where we haven't defined xp before. 

Do we have any example where the namespace gets changed during the course of the execution, because that doesn't seem to be desirable behavior?

Reply:
> Do we have any example where the namespace gets changed during the course of the execution, because that doesn't seem to be desirable behavior?

I haven't come across one. My thought was that it could happen by accident and this accident then becomes hard to detect if we reassign the array namespace in the utility functions. But if I understand correctly then this is a broader problem anyways. So, I have removed this None-check.